cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/internal/query/QueryImpl.java

```diff
@@ -137,15 +137,17 @@ public void close() {
      *  @return Map of indexes in the database.
      */
     @Override
-    public List<Index> listIndexes() {
+    public List<Index> listIndexes() throws QueryException {
         try {
             return dbQueue.submit(new ListIndexesCallable()).get();
         } catch (InterruptedException e) {
-            logger.log(Level.SEVERE,"Failed to list indexes",e);
-            throw new RuntimeException(e);
+            String msg = "Failed to list indexes";
```

Comment:
I don't think we should, it's nice to have the the messages strings right next to the code where we log and throw. Admittedly the duplication is annoying, but it is something that can go away in future when we can use Java 7 language features.